pkg/diskmaker/controllers/lvset/reconcile.go

```diff
@@ -136,20 +136,33 @@ func (r *ReconcileLocalVolumeSet) Reconcile(request reconcile.Request) (reconcil
 		reqLogger.Info("found stale symLink Entries", "storageClass.Name", storageClass, "paths.List", noMatch, "directory", symLinkDir)
 	}
 
-	return reconcile.Result{Requeue: true, RequeueAfter: time.Minute}, nil
+	// shorten the requeueTime if there are delayed devices
+	requeueTime := time.Minute
+	if len(delayedDevices) > 1 {
+		requeueTime = deviceMinAge / 2
+	}
+
+	return reconcile.Result{Requeue: true, RequeueAfter: requeueTime}, nil
 }
 
-// runs filters and matchers on the blockDeviceList and returns valid devices.
+// runs filters and matchers on the blockDeviceList and returns valid devices
+// and devices that are not considered old enough to be valid yet
+// i.e. if the device is younger than deviceMinAge
+// if the waitingDevices list is nonempty, the operator should requeueue
 func (r *ReconcileLocalVolumeSet) getValidDevices(
 	reqLogger logr.Logger,
 	lvset *localv1alpha1.LocalVolumeSet,
 	blockDevices []internal.BlockDevice,
-) []internal.BlockDevice {
+) ([]internal.BlockDevice, []internal.BlockDevice) {
 	validDevices := make([]internal.BlockDevice, 0)
+	delayedDevices := make([]internal.BlockDevice, 0)
 	// get valid devices
 DeviceLoop:
 	for _, blockDevice := range blockDevices {
 
+		// get/set firstObserved
+		firstObserved := r.deviceAgeMap.getFirstObserved(blockDevice.KName)
```

Comment:
it might be cleaner to rename this function as  - `availableAt` so as this function reports time at which device will become available here.

Reply:
hmm, I tried to keep `ageMap` generic. this would compromise that